zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/FlowControl.java

```diff
@@ -181,6 +189,10 @@ public void onProcessed(final long position) {
       inFlightEntry.onProcessed();
     }
     lastProcessedPosition = position;
+    if (writeRate.observe(position) && writeRateLimit != null && writeRateLimit.enabled()) {
+      metrics.setPartitionLoad(
+          Math.min((float) (writeRate.rate() / writeRateLimiter.getRate() * 100L), 100));
+    }
```

Comment:
:x: Just thought of this now but this should actually be part of `onWrite`, not `onProcessed`.